monai/transforms/compose.py

```diff
@@ -358,3 +359,61 @@ def inverse(self, data):
         for o in reversed(applied_order):
             data = apply_transform(self.transforms[o].inverse, data, self.map_items, self.unpack_items, self.log_stats)
         return data
+
+
+class SomeOf(Compose):
+    """
+    ``SomeOf`` provides the ability to randomly sample a fixed number or
+    up to a fixed number of transforms from a list of callables.
+
+    If sampling up to the max_samples, each invocation of SomeOf uniformly samples a target number of transforms
+    for that invocation.
+
+    At each invocation the set of transforms to apply is uniformly sampled without replacement.
+
+    Args:
+        transforms: list of callables.
+        map_items: whether to apply transform to each item in the input `data` if `data` is a list or tuple.
+            Defaults to `True`.
+        unpack_items: whether to unpack input `data` with `*` as parameters for the callable function of transform.
+            Defaults to `False`.
+        log_stats: whether to log the detailed information of data and applied transform when error happened,
+            for NumPy array and PyTorch Tensor, log the data shape and value range,
+            for other metadata, log the values directly. Default to `False`.
+        min_samples: minimum number of transforms to sample. Defaults to `0`.
+        max_samples: maximum number of transforms to sample. Defaults to `3`.
```

Comment:
A little confused about these args, maybe `min_num_transforms` or `min_num`. Do you have some suggestions on the name? @wyli @Nic-Ma

Reply:
Thanks for the comment. I've switched the naming to `min_num_transforms` and `max_num_transforms`. Let me know if you prefer a different name. Will explain those variables in the comment below.